locale/en/docs/guides/backpressuring-in-streams.md

```diff
@@ -0,0 +1,264 @@
+                 |       |        |  > if (!chunk)           |             |
+                 ^       |        |      emit .end();        |             |
+                 ^       ^        |  > else                  |             |
+                 |       ^        |      emit .write();  +---v---+     +---v---+
+                 |       |        ^----^-----------------<  No   |     |  Yes  |
+                 ^       |                               +-------+     +---v---+
+                 ^       |                                                 |
+                 |       ^   emit .pause();        +=================+     |
+                 |       ^---^---------------------+  return false;  <-----+---+
+                 |                                 +=================+         |
+                 |                                                             |
+                 ^   when queue is empty   +============+                      |
+                 ^---^-----------------^---<  Buffering |                      |
+                     |                     |============|                      |
+                     +> emit .drain();     |  <Buffer>  |                      |
+                     +> emit .resume();    +------------+                      |
+                                           |  <Buffer>  |                      |
+                                           +------------+  add chunk to queue  |
+                                           |            <--^-------------------<
+                                           +============+
+```
+
+
+_Note_: The `.pipe()` function is typically where backpressure is invoked. In 
+an isolate application, both [`Readable`][] and [`Writable`][] streams 
+should be present. If you are writing an application meant to accept a 
+[`Readable`][] stream, or pipes to a [`Writable`][] stream from another app, 
+you may omit this detail.
+
+## Example App
+
+Since [Node.js v0.10][], the [`Streams`][] class has offered the ability to 
+overwrite the functionality of [`.read()`][] or [`.write()`][] by using the
+underscore version of these respective functions ([`._read()`][] and 
+[`._write()`][]).
+
+There are guidelines documented for [implementing Readable streams][] and 
+[implementing Writable streams][]. We will assume you've read these over. 
+
+This application will do something very simple: take the data source and 
+perform something fun to the data, and then write it to another file.
+
+## Rules to Abide For Writing Custom Streams 
+
+Recall that a [`.write()`][] may return true or false dependent on some 
+conditions. Thus, when building our own [`Writable`][] stream, we must pay 
+close attention those conditions and the return value:
+
+* If the write queue is busy, [`.write()`][] will return false.
+* If the data chunk is too large, [`.write()`][] will return false (the limit 
+is indicated by the variable, [`highWaterMark`][]).
```

Comment:
If we are _building_ a `Writable` directly, this is not needed, as it is handled by the stream state machine. However, it is needed to **use** a `Writable`  directly, see my example in the Node docs: https://github.com/nodejs/node/blob/master/doc/api/stream.md#writablewritechunk-encoding-callback